File: 1407-group-the-people-given-the-group-size-they-belong-to/1407-group-the-people-given-the-group-size-they-belong-to.py
class Solution:
    def groupThePeople(self, groupSizes: List[int]) -> List[List[int]]:
        group = {}
        for i in range(len(groupSizes)):
            x = groupSizes[i]
            if x in group.keys():
                group[x].append(i)
            else:
                group[x] = [i]
        result = []
        for k,v in group.items():
            for i in range(0,len(v),k):
                result.append(v[i:i+k])
        return result

File: 1407-group-the-people-given-the-group-size-they-belong-to/1407-group-the-people-given-the-group-size-they-belong-to.java
class Solution {
    public List<List<Integer>> groupThePeople(int[] groupSizes) {
        List<List<Integer>> result = new ArrayList<>();
        HashMap<Integer,List<Integer>> hm = new HashMap<>();
        for (int i = 0; i < groupSizes.length; i++) {
            if (hm.containsKey(groupSizes[i])) {
                hm.get(groupSizes[i]).add(i);
            }
            else {
                List<Integer> list = new ArrayList<>();
                list.add(i);
                hm.put(groupSizes[i], list);
            }
        }
        for (int size : hm.keySet()) {
            List<Integer> res = hm.get(size);
            for (int i = 0; i < res.size(); i += size) {
                result.add(res.subList(i,i+size));
            }
        }
        return result;
    }
}